oak/common/utils.cc

```diff
@@ -43,5 +44,46 @@ void write_file(const std::string& data, const std::string& filename) {
   t.close();
 }
 
+std::map<std::string, std::string> read_pem(const std::string& filename) {
+  std::map<std::string, std::string> content;
+  std::ifstream pem_file(filename, std::ifstream::in);
+  if (!pem_file.is_open()) {
+    LOG(FATAL) << "Could not open file '" << filename << "'";
+  }
+
+  std::string line;
+  std::regex header_regex("-----BEGIN (.*)-----");
+  while (std::getline(pem_file, line)) {
+    std::smatch header_match;
+    if (std::regex_search(line, header_match, header_regex)) {
+      std::string header = header_match[1].str();
+      std::string footer = "-----END " + header + "-----";
+      std::string value;
+      // Build the value by concatenating all the lines between the header and the footer.
+      while (std::getline(pem_file, line) && line.find(footer) == std::string::npos) {
+        value.append(line);
+      }
+      content[header] = value;
+    }
+  }
+  return content;
+}
+
+void write_pem(const std::map<std::string, std::string>& map, const std::string& filename) {
+  std::ofstream out_file(filename, std::ofstream::out);
+  if (!out_file.is_open()) {
+    LOG(FATAL) << "Could not open file '" << filename << "'";
+  }
+  for (const auto& item : map) {
+    std::string header = "-----BEGIN " + item.first + "-----";
+    std::string footer = "-----END " + item.first + "-----";
+
+    out_file << header << "\n";
+    out_file << item.second << "\n";
```

Comment:
Shouldn't this also base64 encode / decode? Otherwise it seems it is pretty easy to confuse things if the items contain the string `-----BEGIN` or similar.